tsdb/index/inmem/meta.go

```diff
@@ -395,6 +395,20 @@ func (m *Measurement) TagSets(shardID uint64, opt influxql.IteratorOptions) ([]*
 	return sortedTagsSets, nil
 }
 
+func (m *Measurement) AuthorizeRead(auth influxql.Authorizer) bool {
```

Comment:
I'm concerned about how this is going to perform in some possibly unusual/extreme situations. Apologies if I'm not understanding how the `auth` methods work internally though...

If I understand this properly, this is relying on using unordered iteration of the series on the measurement to find at least one series that we're authorised to read from. So it's the case that a user can read a measurement if they can read at least one series on that measurement, right?

I'm assuming the worse case here would be if we have some tags that we can read, but none of the series on the measurement have those tags? In that case we have to iterate over all the series to determine that we can't read any, and therefore can't read this measurement?

When the number of series for a measurement is large, I'm wondering if we would be better off taking in turn the (likely) small set of tag key/value grants we're interested in and binary searching over the ordered list of series. I realise that's probably too significant a change to the API though...

I guess the type of scenarios I'm thinking of would be something like a user being allowed to read values for a certain container ID (or some other high cardinality data), and the measurement in question containing many container IDs, but not the one the user can read from.